### DISCUSSION TITLE
Need for maintaining eslint-plugin-change-detection-strategy post Angular 16 update

### DISCUSSION THREAD

**Discussion by num13ru** [num13ru/eslint-plugin-change-detection-strategy#6] [Polls]
Hello Everyone 👋,

With the release of Angular 16, we have seen the introduction of a new mechanic - "Computed signals react and recalculate when any of its dependent signals change". This means if a signal is bound in a template, Angular's change detection will automatically update any view that reads the signal whenever the signal changes, allowing the user to see the changed value.

This enhancement has led us to reconsider the need for the "eslint-plugin-change-detection-strategy" that checks if Angular ChangeDetectionStrategy is OnPush. 

Here are the key reasons that Angular gives for the use of signals:

1. **More reactivity:** Signals provide a more responsive experience.
2. **Finer control over change detection:** Using signals can improve performance as they allow us more control over change detection.

In light of these updates, the question that we need to discuss and decide upon is: 

"Do we still need to maintain the 'eslint-plugin-change-detection-strategy'?" 

Do you believe the plugin provides value that can't be achieved by Angular 16's new mechanic? Or do you think the benefits of the new mechanic render the plugin redundant?

Please cast your vote in the poll below and share your thoughts in the comments section. Your input is vital in helping us determine the best way forward for this project.

Looking forward to your valuable insights.

Best,

[Your Name]

**Comment by strigefleur** [comment]
It will take years for all the existing projects to migrate to Angular signals. In the mean time you provide a tool to control app component code style explicitly - so let it fulfill it's purpose.

**Comment by num13ru** [comment]
Okay got you
